consensus/src/chained_bft/safety/safety_rules.rs

```diff
@@ -206,6 +232,12 @@ impl<T: Payload> SafetyRules<T> {
                 }
             }
         }
+
```

Comment:
Are these lines duplicate of the logic we have in the lines above? I'd feel more comfortable if we'd not change business logic in this PR, may be these lines are not necessary here.

Reply:
We can probably stack another commit on top of this to make review easier while keep it in the same pr.

Reply:
I'll remove them and keep this pr clean for now. Will do this when we remove hashtree.